implementations/rust/ockam/ockam_api/src/kafka/inlet_map.rs

```diff
@@ -0,0 +1,144 @@
+use core::str::FromStr;
+use minicbor::Decoder;
+
+use ockam::compat::tokio::sync::Mutex;
```

Comment:
I don't  think I can without risking a deadlock, I'm keeping the lock while exchanging messages to keep the locking mechanism simpler, and if another task tries to lock while waiting for a message it could cause a deadlock (if tokio assigned the same thread to both tasks or if we are running single-threaded)

Reply:
I don't think there can be a deadlock. Sync mutex lock won't allow you to carry it across async calls

Reply:
> I don't think there can be a deadlock. Sync mutex lock won't allow you to carry it across async calls

You're right, the compiler stop me from doing it! Should I change the locking schema to use sync mutex?

Reply:
That would be great!